packages/snaps-utils/src/snaps.ts

```diff
@@ -230,27 +232,47 @@ export function validateSnapShasum(
   }
 }
 
+/**
+ * Sanitizes a potential snap id string.
+ *
+ * @param snapId - A potential snap id.
+ * @returns A sanitized potential snap id.
+ */
+function sanitizeSnapId(snapId: string) {
```

Comment:
Instead of sanitising we should outright reject them. A whitespace in an ID means there's an error, either in our logic or the user set-up, that error shouldn't bubble up silently.

Reply:
Fair enough 👍